### DISCUSSION TITLE
Display OLED 0.96" not working after upgrade from 13.4 to 14.x

### DISCUSSION THREAD

**Discussion by gkellershs** [arendst/Tasmota#22525] [General]
I did an OTA Upgrade on an "ESP8266-D1mini" from version 13.4 to 14.x an didn´t get the display working.

display shows

`RSL: RESULT = {"Display":{"Model":2,"Type":0,"Width":128,"Height":64,"Mode":1,"Dimmer":50,"Size":2,"Font":2,"Rotate":0,"Invert":0,"Refresh":1,"Cols":[16,8],"Rows":2}}`

and i2cscan shows 

`RSL: RESULT = {"I2CScan":"Device(s) found at 0x3c"}`

Running on version 14.x I got displaymodel  = 17 and could not change to Model=2

Did I something wrong, do I have to change something for the upgrade. I´m using Tasmota 13.4.0(display)

Thanks for help, Georg

**Comment by sfromis** [comment]
With Tasmota version 14, the display support has been upgraded to use Universal Display, which you have to configure a bit differently. You need a display descriptor, and configuring some "random" unused gpio as `Option A 3`. 
https://tasmota.github.io/docs/Universal-Display-Driver/#migrating-to-udisplay

With SSD1306 and `tasmota-display`, the simplest setup is probably to use rule set 3 for the display descriptor.
```
Rule3 :H,SSD1306,128,64,1,I2C,3c,*,*,* :S,0,2,1,0,30,20 :I AE D5,80 A8,3F D3,00 40 8D,14 20,00 A1 C8 DA,12 81,9F D9,F1 DB,40 A4 A6 AF :o,AE :O,AF :A,00,10,40,00,00 :i,A6,A7 #
```

**Reply by Gurugismo** [reply]
thanks his tip was good for me

**Comment by gkellershs** [comment]
solved by answer from sfromis
